raiden_contracts/tests/test_channel_update_transfer.py

```diff
@@ -353,6 +356,471 @@ def test_update_channel_fail_no_offchain_transfers(
+
+    # Later balance proof, higher transferred amount, higher nonce
+    balance_proof_B_1 = create_balance_proof(channel_identifier, B, 10, 0, 4, fake_bytes(32, '02'))
+
+    # B's signature on the update message is valid
+    balance_proof_update_signature_B = create_balance_proof_update_signature(
+        B,
+        channel_identifier,
+        *balance_proof_B_1,
+    )
+
+    # A closes with the fist balance proof
+    token_network.functions.closeChannel(B, *balance_proof_B_0).transact({'from': A})
+
+    # Someone wants to update with later balance proof - not possible
+    with pytest.raises(TransactionFailed):
+        token_network.functions.updateNonClosingBalanceProof(
+            A,
+            B,
+            *balance_proof_B_1,
+            balance_proof_update_signature_B,
+        ).transact({'from': A})
+    with pytest.raises(TransactionFailed):
+        token_network.functions.updateNonClosingBalanceProof(
+            A,
+            B,
+            *balance_proof_B_1,
+            balance_proof_update_signature_B,
+        ).transact({'from': B})
+    with pytest.raises(TransactionFailed):
+        token_network.functions.updateNonClosingBalanceProof(
+            A,
+            B,
+            *balance_proof_B_1,
+            balance_proof_update_signature_B,
+        ).transact({'from': M})
+
+
+def test_updateNonClosingBalanceProof_invalid_BP_arguments(
+        token_network,
+        token_network_test,
+        create_channel,
+        channel_deposit,
+        get_accounts,
+        create_balance_proof,
+        create_balance_proof_update_signature,
+):
+    (A, B, C) = get_accounts(3)
+    settle_timeout = TEST_SETTLE_TIMEOUT_MIN
+    deposit_A = 20
+    channel_identifier = create_channel(A, B, settle_timeout)[0]
+    channel_deposit(A, deposit_A, B)
+
+    token_network.functions.closeChannel(
+        B,
+        fake_bytes(32),
+        0,
+        fake_bytes(32),
+        fake_bytes(64),
+    ).transact({'from': A})
+
+    BP = namedtuple('BP', ['balance_hash', 'nonce', 'additional_hash', 'signature'])
+
+    #  Create valid balance_proof
+    BP_valid = BP(*create_balance_proof(
+        channel_identifier,
+        A,
+        10,
+        0,
+        2,
+        fake_bytes(32, '02'),
+    ))
+
+    # And a valid nonclosing_signature
+    valid_balance_proof_update_signature = create_balance_proof_update_signature(
+        B,
+        channel_identifier,
+        *BP_valid,
+    )
+
+    #  We test invalid balance proof arguments with valid signatures
+
+    #  Create balance_proof for invalid token_network
+    BP_invalid_token_network = BP(*create_balance_proof(
+        channel_identifier,
+        A,
+        10,
+        0,
+        2,
+        fake_bytes(32, '02'),
+        other_token_network=token_network_test,
+    ))
+
+    signature_invalid_token_network = create_balance_proof_update_signature(
+        B,
+        channel_identifier[::-1],
```

Comment:
From what I understand, you are testing that update fails if someone is using another `token_network_address` - by using the `token_network_test`.
Why add `channel_identifier[::-1]` as another failure point?

Reply:
thx for the catch